src/RamTransforms.cpp

```diff
@@ -641,84 +641,95 @@ bool TupleIdTransformer::reorderOperations(RamProgram& program) {
 bool HoistAggregateTransformer::hoistAggregate(RamProgram& program) {
     bool changed = false;
 
-    // hoist a single aggregate to an outer scope
-    // most outer permissible aggregate is hoisted
+    // There are two cases: aggregates that have no data-dependencies on
+    // other RAM operations, and aggregates that have data-dependencies.
+
+    // Hoist a single aggregate to an outer scope that
+    // has no data-dependency.
+    visitDepthFirst(program, [&](const RamQuery& query) {
+        std::unique_ptr<RamNestedOperation> newAgg;
+        int lastTupleOp = -1;
```

Comment:
This is only used to identify if a non-aggregate tuple operation has been found. Could this instead be a boolean?

Reply:
I renamed it.